### DISCUSSION TITLE
IO7 LED on SonOff USB Dongle

### DISCUSSION THREAD

**Discussion by oliv3r** [Koenkk/Z-Stack-firmware#508] [Q&A]
Hey all,

I noticed in the coordinator patches, that code was added for the 'green led' or the led connected to IO7.

I've soldered the LED (Only had an 0805, so soldered that) and a 1k 0603. With my multimeter, that has an integrated LED tester (e.g. the led lights up) when I connect my MM to the IO7 pad at the bottom, and to ground, I see that my soldering skills have worked and the led lights up.

However, I don't see it doing anything in software. I figured the firmware doesn't care, it just assumes IO7 is either fitted with a LED or not, so it's always enabled.

Does anybody have a device with IO7 that has an LED and what is expected to happen?

I tried zigpy-cli only so far, and restored my json and performed an energyscan. Also BSL flashing doesn't produce anything on the LED. I checked whether it was simply to dimm/overshadowed by the RED led, but that doesn't seem to be the case either.

I know it'll end up within the metal case, so I won't ever see it. But still curious what the LED was for, and if it was supposed to work ...

**Comment by oliv3r** [comment]
So, from code I see that there's two use-cases. One is via the API (0x0A) so sending a message can turn it on/off it seems.

The other use cases is that the LED is toggled when Joining of the network is permitted (and timed out).

Anyway is this assumtion true? anybody know? @koenkk? When I use `zigpy ... permit` the LED seems to stay off :(

**Reply by Koenkk** [reply]
I wasnt aware the dongle P has a led, are we using the correct pin number in the fw?

**Comment by oliv3r** [comment]
Illvdouble check what CONFIG_LED_GREEN maps too, but the silkscreen has it as IO7. Also by default it doest. I soldered it o.. pic tomorrow :p

**Reply by oliv3r** [reply]
So, the LED seems to be configured properly; https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/firmware.patch#L1077 it is indeed IO7.

If we look at https://www.ti.com/document-viewer/CC2652P/datasheet#GUID-8654399D-F60D-4CFD-828C-02C6620DB954/TITLE-SWRS192TOCWRAPPER_TERMINAL_CONFIGURATION_AND_FUNCTIONS we see that IO7 is the bottom left pin.

Tracing it out, it pops to the bottom to the testpoint marked IO7. Following the trace brings us to the led. The LED lights up with I apply a little voltage on the test pad, so it works.

![2024-08-16-09-11-07-096](https://github.com/user-attachments/assets/42542fb5-f28a-4529-9645-4ae2ba142ac6)
![2024-08-16-09-11-18-440](https://github.com/user-attachments/assets/077fcff9-fcbc-455c-8f06-4faf24fdf662)
